samples/subsys/usb/hid-cdc/src/main.c

```diff
@@ -0,0 +1,868 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <device.h>
+#include <gpio.h>
+#include <uart.h>
+#include <string.h>
+#include <random/rand32.h>
+
+#include <usb/usb_device.h>
+#include <usb/class/usb_hid.h>
+#include <usb/class/usb_cdc.h>
+
+#define LOG_LEVEL LOG_LEVEL_DBG
+LOG_MODULE_REGISTER(main);
+
+#ifdef SW0_GPIO_CONTROLLER
+#define PORT0 SW0_GPIO_CONTROLLER
+#else
+#error SW0_GPIO_CONTROLLER needs to be set
+#endif
+
+#ifdef SW0_GPIO_PIN
+#define PIN0     SW0_GPIO_PIN
+#else
+#error SW0_GPIO_PIN needs to be set
+#endif
+
+#ifdef SW0_GPIO_FLAGS
+#define PIN0_FLAGS SW0_GPIO_FLAGS
+#else
+#error SW0_GPIO_FLAGS needs to be set
+#endif
+
+#ifdef SW1_GPIO_PIN
+#define PIN1	SW1_GPIO_PIN
+#endif
+
+#ifdef SW1_GPIO_CONTROLLER
+#define PORT1	SW1_GPIO_CONTROLLER
+#endif
+
+#ifdef SW1_GPIO_FLAGS
+#define PIN1_FLAGS SW1_GPIO_FLAGS
+#endif
+
+#ifdef SW2_GPIO_PIN
+#define PIN2	SW2_GPIO_PIN
+#endif
+
+#ifdef SW2_GPIO_CONTROLLER
+#define PORT2	SW2_GPIO_CONTROLLER
+#endif
+
+#ifdef SW2_GPIO_FLAGS
+#define PIN2_FLAGS SW2_GPIO_FLAGS
+#endif
+
+#ifdef SW3_GPIO_PIN
+#define PIN3	SW3_GPIO_PIN
+#endif
+
+#ifdef SW3_GPIO_CONTROLLER
+#define PORT3	SW3_GPIO_CONTROLLER
+#endif
+
+#ifdef SW3_GPIO_FLAGS
+#define PIN3_FLAGS SW3_GPIO_FLAGS
+#endif
+
+#define LED_PORT	LED0_GPIO_CONTROLLER
+#define LED		LED0_GPIO_PIN
+
+/* Event FIFO */
+
+K_FIFO_DEFINE(evt_fifo);
+
+enum evt_t {
+	GPIO_BUTTON_0	= 0x00,
```

Comment:
Why are you explicitly numbering the member of this enumeration, is their value important? If so please document it.

Reply:
It's not, I just prefer it that way.